src/client.ts

```diff
@@ -122,6 +122,9 @@ export default class Client {
 
   publish(stream: LocalStream) {
     stream.publish(this.transports[Role.pub].pc);
+    setTimeout(() => {
+      this.onNegotiationNeeded();
+    }, 2000);
```

Comment:
is this necessary?

Reply:
don’t know why, I tested it locally and after stream.publish was run, onNegotiationNeeded was not triggered, so I added this test code.

Reply:
hmm can you consistently reproduce? what browser?

Reply:
I used chrome 87.0.4280.67 (x86_64) on macOS 10.15.7 (19H15)

Reply:
Yes, it can always be reproduced

Reply:
can you outline steps to reproduce? is it possible with any of the existing `ion-sfu` examples?

Reply:
run ion-sfu/grpc-web.
```bash
git clone https://github.com/pion/ion-sfu
cd ion-sfu && git checkout feat/grpc-web
go run cmd/signal/grpc/grpc-web/main.go --allow_all_origins --use_websockets --c config.toml
```

use patched echotest.

* `cd ion-sdk-js && npm run bundle`
* copy `dist` to `ion-sfu/examples/echotest` from `ion-sdk-js/dist`

```patch
diff --git a/examples/echotest/index.html b/examples/echotest/index.html
index af8cd0a..7f63985 100644
--- a/examples/echotest/index.html
+++ b/examples/echotest/index.html
@@ -328,7 +328,7 @@
       integrity="sha384-B4gt1jrGC7Jh4AgTPSdUtOBvfO8shuf57BaghqFfPlYxofvL8/KUEfYiJOMMV+rV"
       crossorigin="anonymous"
     ></script>
-    <script src="https://unpkg.com/ion-sdk-js@1.5.0/dist/ion-sdk.min.js"></script>
+    <script src="dist/ion-sdk.min.js"></script>
     <script>
       const localVideo = document.getElementById("local-video");
       const remoteVideo = document.getElementById("remote-video");
@@ -350,11 +350,11 @@
       /* eslint-env browser */
       const joinBtns = document.getElementById("start-btns");
 
-      const signalLocal = new IonSDK.IonSFUJSONRPCSignal(
-        "ws://localhost:7000/ws"
+      const signalLocal = new IonSDK.IonSFUGRPCGWebSignal(
+        "http://localhost:9090"
       );
-      const signalRemote = new IonSDK.IonSFUJSONRPCSignal(
-        "ws://localhost:7000/ws"
+      const signalRemote = new IonSDK.IonSFUGRPCGWebSignal(
+        "http://localhost:9090"
       );
 
       const clientLocal = new IonSDK.Client("test session", signalLocal);

```

Reply:
@tarrencev fixed https://github.com/pion/ion-sdk-js/commit/ac0d9933dd74ee93533036ffe383fb9b10dd0e08